lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1206,7 +1206,8 @@ def _build_files_manifest_walk(b_collection_path, namespace, name, ignore_patter
     def _walk(b_path, b_top_level_dir):
         for b_item in os.listdir(b_path):
             b_abs_path = os.path.join(b_path, b_item)
-            b_rel_base_dir = b'' if b_path == b_top_level_dir else b_path[len(b_top_level_dir) + 1:]
+            common_prefix = os.path.commonprefix([b_top_level_dir, b_path])
```

Comment:
[suggestion (blocking)][conventional comments]: No need to evaluate this on each loop iteration, it's cheaper to have it calculated outside the loop.

[conventional comments]: https://conventionalcomments.org/

Reply:
done, testing

Reply:
Marking this thread as resolved.